### DISCUSSION TITLE
Build Captive Air sdk application in continuous integration environment

### DISCUSSION THREAD

**Discussion by pierrehub2b** [airsdk/Adobe-Runtime-Support#2763] [Q&A]
Hello,

We build all our applications and components with Jenkins and GitLab.
How can we retrieve automatically the last version of the Harman Flex Sdk archive ?

Pierre

**Comment by ajwfrost** [comment]
Hi

I would suggest that it may be better to manually make a decision to update.. you won't necessarily want to have the latest SDK particularly when major changes are made and there's a higher risk of regression - this is one of the reasons why we've adjusted the version number approach and why the SDK Manager treats separate a.b.c versions as different installs, but the final number is used as the patch level and those should only be low-risk, minor changes..

In terms of programmatically accessing things, this is potentially possible through some APIs but it would probably be better if we added a command-line capability to the AIR SDK Manager. Such that you could script something there e.g. have a configuration setting for the a.b.c.x version that you're happy with, and get the SDK Manager to automatically check for and update any a.b.c.(x+1) versions...

For the short term, the simplest option would be to have the AIR SDK Manager on your build node and then manually use this to keep a particular location up to date.. but let me know what you think about the potential for command-line usage, and what commands might be useful there.

thanks

**Reply by pierrehub2b** [reply]
Hello Andrew, thank you for your very fast response.
Today we are using Jenkins and Gitlab CI to build our componants (Jenkins master is a Linux OS and we have Jenkins Agent on windows VMs), and in a next step we want to build our componants and applications with Docker too.
So I think the best solution for us is to define an AirSDK version in a properties file and automatically download the SDK if the version is not present on the build machine.
I am ok if the prerequisite is to install AirSDKManager and I think the command-line usage will be a great evolution for our needs.

Thanks again Andrew
